packages/schematics/angular/utility/ast-utils.ts

```diff
@@ -568,3 +579,89 @@ export function isImported(source: ts.SourceFile,
 
   return matchingNodes.length > 0;
 }
+
+/**
+ * Returns the RouterModule declaration from NgModule metadata, if any.
+ */
+export function getRouterModuleDeclaration(source: ts.SourceFile): ts.Expression | undefined {
+  const result = getDecoratorMetadata(source, 'NgModule', '@angular/core') as ts.Node[];
+  const node = result[0] as ts.ObjectLiteralExpression;
+  const matchingProperties = getMetadataField(source, node, 'imports');
+
+  if (!matchingProperties) {
+    return;
+  }
+
+  const assignment = matchingProperties[0] as ts.PropertyAssignment;
+
+  if (assignment.initializer.kind !== ts.SyntaxKind.ArrayLiteralExpression) {
+    return;
+  }
+
+  const arrLiteral = assignment.initializer as ts.ArrayLiteralExpression;
+
+  return arrLiteral.elements
+    .filter(el => el.kind === ts.SyntaxKind.CallExpression)
+    .find(el => (el as ts.Identifier).getText(source).startsWith('RouterModule'));
+}
+
+/**
+ * Adds a new route declaration to a router module (i.e. has a RouterModule declaration)
+ */
+export function addRouteDeclarationToModule(
+  source: ts.SourceFile,
+  fileToAdd: string,
+  routeLiteral: string,
+): Change {
+  const routerModuleExpr = getRouterModuleDeclaration(source);
+  if (!routerModuleExpr) {
+    throw new Error(`Couldn't find a route declaration.`);
+  }
+  const scopeConfigMethodArgs = (routerModuleExpr as ts.CallExpression).arguments;
+  if (!scopeConfigMethodArgs.length) {
+    throw new Error(`The router module method doesn't have arguments.`);
+  }
+
+  let routesArr: ts.ArrayLiteralExpression | undefined;
+  const routesArg = scopeConfigMethodArgs[0];
+
+  // Check if the route declarations array is
+  // an inlined argument of RouterModule or a standalone variable
+  if (routesArg.kind === ts.SyntaxKind.ArrayLiteralExpression) {
+    routesArr = routesArg as ts.ArrayLiteralExpression;
+  } else {
+    const routesVarName = routesArg.getText();
+    const routesVar = source.statements
+      .filter((s: ts.Statement) => s.kind === ts.SyntaxKind.VariableStatement)
+      .find((v: ts.VariableStatement) => {
+        return v.declarationList.declarations[0].name.getText() === routesVarName;
+      }) as ts.VariableStatement | undefined;
+
+    if (!routesVar) {
+      throw new Error(`The route declaration variable "${routesVarName}" is used but not defined.`);
```

Comment:
Can the error provide file and position information as well?  Same for others where applicable.

Reply:
I'll see what I can do about that. :+1:

Reply:
I assume the line of the error and the file name will be enough? Intentionally omitted the column/character position since I think that it may just add an additional noise.